WalletWasabi.Tests/UnitTests/BlockchainAnalysis/PubKeyReuseAnonScoreTests.cs

```diff
@@ -35,6 +35,45 @@ public void AddressReusePunishment()
 			Assert.True(tx.WalletOutputs.First().HdPubKey.AnonymitySet < 30);
 		}
 
+		[Fact]
+		public void AddressReusePunishmentProcessTwice()
+		{
+			// If there's reuse in input and output side, then output side didn't gain, nor lose anonymity.
+			var analyser = ServiceFactory.CreateBlockchainAnalyzer();
+			var km = ServiceFactory.CreateKeyManager();
+			var reuse = BitcoinFactory.CreateHdPubKey(km);
+			var tx = BitcoinFactory.CreateSmartTransaction(
+				9,
+				Enumerable.Repeat(Money.Coins(1m), 9),
+				new[] { (Money.Coins(1.1m), 100, BitcoinFactory.CreateHdPubKey(km)) },
+				new[] { (Money.Coins(1m), HdPubKey.DefaultHighAnonymitySet, reuse) });
+
+			// Make the reused key anonymity set something smaller than 109 (which should be the final anonymity set)
+			reuse.SetAnonymitySet(30, uint256.One);
+
+			analyser.Analyze(tx);
+			var inputAnonsets = tx.WalletInputs.Select(x => x.HdPubKey.AnonymitySet).ToArray();
+			var outputAnonsets = tx.WalletOutputs.Select(x => x.HdPubKey.AnonymitySet).ToArray();
+
+			analyser.Analyze(tx);
+			var newInputAnonsets = tx.WalletInputs.Select(x => x.HdPubKey.AnonymitySet).ToArray();
+			var newOutputAnonsets = tx.WalletOutputs.Select(x => x.HdPubKey.AnonymitySet).ToArray();
+
+			// Anonsets should not change.
+			for (int i = 0; i < inputAnonsets.Length; i++)
+			{
+				var anonset = inputAnonsets[i];
+				var newAnonset = newInputAnonsets[i];
+				Assert.Equal(anonset, newAnonset);
+			}
```

Comment:
You can use `Assert.Equal` even for arrays thanks to https://github.com/xunit/assert.xunit/pull/6.

Tested with:

```csharp
Assert.Equal<int[]>(new int[] { 1, 2, 3 }, new int[] { 2, 3 });
```

which gives you:

```
    Assert.Equal() Failure
    Expected: Int32[] [1, 2, 3]
    Actual:   Int32[] [2, 3]
 ```

Reply:
Done